tutorials/plugins/gdnative/files/cpp_example/SConstruct

```diff
@@ -1,48 +1,92 @@
 #!python
-import os
+import os, subprocess
 
-target = ARGUMENTS.get("target", "debug")
-platform = ARGUMENTS.get("platform", "windows")
-bits = ARGUMENTS.get("bits", 64)
+opts = Variables([], ARGUMENTS)
 
-final_lib_path = 'demo/bin/'
+# Gets the standard flags CC, CCX, etc.
+env = DefaultEnvironment()
 
-# This makes sure to keep the session environment variables on windows,
-# that way you can run scons in a vs 2017 prompt and it will find all the required tools
-env = Environment()
-if platform == "windows":
-    env = Environment(ENV = os.environ)
+# Define our options
+opts.Add(EnumVariable('target', "Compilation target", 'debug', ['d', 'debug', 'r', 'release']))
+opts.Add(EnumVariable('platform', "Compilation platform", 'windows', ['windows', 'x11', 'linux', 'osx']))
+opts.Add(EnumVariable('p', "Compilation target, alias for 'platform'", 'not set', ['windows', 'x11', 'linux', 'osx', 'not set']))
```

Comment:
I would keep things consistent with Godot itself and have both `platform` and `p` default to an empty string (so you're forced to use a platform argument and be explicit about what you're doing), and drop the `not set` option here (just check `env['p'] != ''` below).

Reply:
Will do :)